components/tidb_query_expr/src/impl_json.rs

```diff
@@ -159,10 +159,47 @@ pub fn json_merge(args: &[Option<JsonRef>]) -> Result<Option<Json>> {
 #[rpn_fn(writer)]
 #[inline]
 fn json_quote(input: BytesRef, writer: BytesWriter) -> Result<BytesGuard> {
-    let str = String::from_utf8(input.to_vec()).map_err(tidb_query_datatype::codec::Error::from)?;
-    let result =
-        Bytes::from(serde_json::to_string(&str).map_err(tidb_query_datatype::codec::Error::from)?);
-    Ok(writer.write(Some(result)))
+    Ok(writer.write(quote(input)?))
+}
+
+fn quote(bytes: BytesRef) -> Result<Option<Bytes>> {
+    let mut result = Vec::with_capacity(bytes.len() * 2 + 2);
+    result.push(b'\"');
+    for byte in bytes.iter() {
+        if *byte == b'\"' || *byte == b'\\' {
+            result.push(b'\\');
+            result.push(*byte)
+        } else if *byte == b'\x07' {
```

Comment:
Please take https://dev.mysql.com/doc/refman/8.0/en/json-modification-functions.html#json-unquote-character-escape-sequences a look.

Reply:
yep, but it seems to be different from `strconv` [here](https://golang.org/src/strconv/quote.go?s=2262:2273#L82)

Reply:
OK, Let's keep the same with TiDB for now.